pkg/route/ingress/ingress.go

```diff
@@ -882,3 +889,22 @@ func terminationPolicyForIngress(ingress *networkingv1.Ingress) routev1.TLSTermi
 		return routev1.TLSTerminationEdge
 	}
 }
+
+var destinationCACertificateSecretKeyName = "destinationCACertificate"
+var destinationCACertificateAnnotationKey = routev1.GroupName + "/destinationCACertificateSecret"
+
+func destinationCACertificateForIngress(ingress *networkingv1.Ingress, secretLister corelisters.SecretLister) *string {
+	name := ingress.Annotations[destinationCACertificateAnnotationKey]
+	secret, err := secretLister.Secrets(ingress.Namespace).Get(name)
+	if err != nil {
+		return nil
+	}
+	if secret.Type != corev1.SecretTypeOpaque {
+		return nil
+	}
+	if v, ok := secret.Data[destinationCACertificateSecretKeyName]; ok {
```

Comment:
Is there a reason not to use the more standard `tls.crt` as the data key name?  We should indeed allow using `Opaque`-type secrets so that the user isn't forced to provide both `tls.crt` and `tls.key` when we only need the former.  However, we might as well also allow using `kubernetes.io/tls`-type secrets (though we don't use the `tls.key`), which some users may prefer and already be using to store the destination CA certificate.  

```suggestion
	switch secret.Type {
	case corev1.SecretTypeTLS, corev1.SecretTypeOpaque:
	default:
		return nil
	}
	if v, ok := secret.Data[corev1.TLSCertKey]; ok {
```

Note also that `NewController` sets up a watch on secrets with a filter that excludes `Opaque`-type secrets.  If we want the ingress-to-route controller to update the route when the secret that the ingress references is modified, then we will need to change that.  We should also add a test case for the same, if we decide to implement it.   However, maybe we can leave updating when the referenced secret changes as a follow-up change if it's too complex to do in this PR.

Reply:
I've added this functionality, now the filter will considers the Opaque type secrets and a Secret type switch could happen (Added a test case too).